### ISSUE TITLE
Problem with the connectivity

### ISSUE THREAD

**Issue by aaronariasperez** [CymatiCorp/CyKITv2#4]
Hi, I have a problem when I try to connect the controller.
When I execute the example 1,  python.exe CyKITv2.py 127.0.0.1 18675 2 , I get this error:

Traceback (most recent call last):
  File "C:\Python27\lib\threading.py", line 810, in __bootstrap_inner
    self.run()
  File "C:\Python27\lib\threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "C:\Users\SIC\Downloads\CyKITv2-master\Python\eeg.py", line 239, in run
    cipher = AES.new(key, AES.MODE_ECB, iv)
  File "C:\Python27\lib\site-packages\Crypto\Cipher\AES.py", line 202, in new
    return _create_cipher(sys.modules[__name__], key, mode, *args, **kwargs)
  File "C:\Python27\lib\site-packages\Crypto\Cipher\__init__.py", line 53, in _c
reate_cipher
    raise TypeError("IV is not meaningful for the ECB mode")
TypeError: IV is not meaningful for the ECB mode

What is the problem? Thanks.

**Comment by warrenarea** [comment]
Find out what version of pycrypto you are using. 

That 'might' be a new standard for pycrypto, where they might be getting rid of it.
either that, you might also just need to reinstall it.

finding the version would be the first step.

**Comment by aaronariasperez** [comment]
I have the 2.6.1 version of pycrypto so I don't know where is the problem.

I don't know if these problems are caused by the pycrypto problem:
On the browser I get the 'Connected' status, but only when I connect the EPOC with the usb wire.
I get a 'None' in the controller.

Thanks for the help.

**Comment by warrenarea** [comment]
yeah, you can't connect to it directly using USB, i am looking to add that feature eventually.
will probably create a "Settings" tab.

I think there is definitely a problem with your pycrypto library, you might try uninstalling it,
and reinstalling it.   you might also try removing the ", IV" from the eeg.py and see if that works,
otherwise, i think you have to find an install of pycrypto that works.

let me know if you get one to work.

**Comment by warrenarea** [comment]
cipher = AES.new(key, AES.MODE_ECB, iv)

basically that line is giving you the problem
you could "try" changing it to 
cipher = AES.new(key, AES.MODE_ECB)

but i can't guarantee that will work... they have obviously changed something.

other than that, i can only recommend you reinstall it, or try reinstalling a previous
version, maybe even 2.6

**Comment by warrenarea** [comment]
if that doesn't work, try on line 239~
under the run(self, key, myio)  function

print str(key)
and try to connect again.... perhaps i missed your product name, and it just
doesn't have a key to work with.

if you want to check on your own, have it print out the product names, run CyKITv2 as

CyKITv2.py 127.0.0.1 18675 2 info+confirm

**Comment by aaronariasperez** [comment]
thanks for the response, I have made it work, but I don't get any data using the auto mode and it should print the data from {AF3, F7, F3, FC5, T7....}. 
I get some data from some sensors when I select it manually, but I'm not sure if those datas are correct.

**Comment by warrenarea** [comment]
yes, for the Epoc+ you have to use the "manual" setting.

I was still working out which sensors were which.

The data in Epoc+ comes in packets in this manner:

[counter] 
[interpolate] (16 for data and 32 for gyro ie MEMS)
[possible data or raw contact quality]
[sensor_data_1]  127
[sensor_data_2]
[sensor_data_1]  127
[sensor_data_2]
[sensor_data_1]  127
[sensor_data_2]
[sensor_data_1]  127
[sensor_data_2]
[sensor_data_1]  127
[sensor_data_2]
0 0 
[sensor_data_1]  127
[sensor_data_2]
[sensor_data_1]  127
[sensor_data_2]
[sensor_data_1]  127
[sensor_data_2]
[sensor_data_1]  127
[sensor_data_2]

where the headset, in its default neutral state, (not being worn) will 
adjust the sensor_data_1 to be 127

and it fluctuates above or below 127, based on whether the sensor impedance
is positive or negative.

so basically, from what i'm gathering is that the 127 sensor_data_1 is what is being
used primarily for the 4200uv sensor value. (the decimal value)

and then the.. sensor_data_2 varies a lot... which i suspect is the high-precision Float data.

I'm just not certain what math they're using, and I'm trying to get some help to determine
how its done... I have some correlated data... just not sure how it all fits together.

**Comment by aaronariasperez** [comment]
I'm very interested in that, I would like to get the data to use them as I want.  I mean, I want to make a trainer without using their software... I will be attentive to your project.

**Comment by warrenarea** [comment]
https://discord.gg/gTYNWc7
If anyone would like to assist me in the Epoc+ data
I have a Discord chat server setup here.  

You can either download the discord chat messenger, 
or just click the link to use the browser interface.

**Comment by warrenarea** [comment]
New version update.
